core/src/main/java/com/exadel/etoolbox/linkinspector/core/services/data/impl/DataFeedServiceImpl.java

```diff
@@ -142,59 +118,63 @@ public List<Resource> dataFeedToResources() {
-        JSONArray jsonArray = JsonUtil.getJsonArrayFromFile(JSON_FEED_PATH, resourceResolver);
-        int allItemsSize = jsonArray.length();
-        if (allItemsSize > 0) {
-            int limit = limited ? Math.min(allItemsSize, UI_ITEMS_LIMIT) : allItemsSize;
-            for (int i = 0; i < limit; i++) {
-                try {
-                    JSONObject jsonObject = jsonArray.getJSONObject(i);
-                    Optional.ofNullable(JsonUtil.jsonToModel(jsonObject, GridResource.class))
-                            .ifPresent(gridResources::add);
-                } catch (JSONException e) {
-                    LOG.error("Failed to convert json object to GridResource", e);
-                }
-            }
+    @Override
+    public void modifyCsvRecord(ResourceResolver resourceResolver, String filePath, Consumer<GridResource> modifyConsumer) {
+        Optional<InputStream> inputStream = Optional
+                .ofNullable(resourceResolver.getResource(filePath))
+                .map(resource -> resource.adaptTo(InputStream.class));
+        if (inputStream.isPresent()) {
+
+            List<GridViewItem> recordDtoList = CsvUtil.readCsvItems(inputStream.get(), CsvUtil.CSV_COLUMNS).stream()
+                    .map(this::parseGridResourceFromCsv)
+                    .peek(modifyConsumer)
+                    .map(gridResource -> toSlingResource(gridResource, resourceResolver))
+                    .map(resource -> resource.adaptTo(GridViewItem.class))
+                    .filter(Objects::nonNull)
+                    .collect(Collectors.toList());
+
+            byte[] csvContentBytes = CsvUtil.itemsToCsvByteArray(recordDtoList, this::printViewItemToCsv, null);
+            LinkInspectorResourceUtil.removeResource(filePath, resourceResolver);
+            LinkInspectorResourceUtil.saveFileToJCR(filePath, csvContentBytes,
+                    CsvUtil.CSV_MIME_TYPE, resourceResolver);
         }
-        return gridResources;
     }
 
-    private void gridResourcesToDataFeed(Collection<GridResource> gridResources, ResourceResolver resourceResolver) {
-        try {
-            JSONArray resourcesJsonArray = JsonUtil.objectsToJsonArray(gridResources);
-            removePreviousDataFeed(resourceResolver);
-            saveGridResourcesToJcr(resourceResolver, resourcesJsonArray);
-            removePendingNode(resourceResolver);
-            resourceResolver.commit();
-            LOG.debug("Saving data feed json to jcr completed, path {}", JSON_FEED_PATH);
-        } catch (PersistenceException e) {
-            LOG.error("Saving data feed json to jcr failed", e);
+    private List<List<GridResource>> dataFeedToGridResources(ResourceResolver resourceResolver) {
+        List<List<GridResource>> gridResources = new ArrayList<>();
+        Resource resource = resourceResolver.getResource(CSV_REPORT_NODE_PATH);
+        if (resource == null) {
+            LOG.error("Resource " + CSV_REPORT_NODE_PATH + " doesn't exist.");
```

Comment:
Recommended notation for Slf4j is `LOG.error("Resource {} doesn't exist", CSV_REPORT_NODE_PATH);`